ceph/datadog_checks/ceph/ceph.py

```diff
@@ -205,21 +226,29 @@ def _extract_metrics(self, raw, tags):
         except KeyError:
             self.log.debug('Error retrieving pgstatus metrics')
 
-        try:
-            num_mons = len(raw['mon_status']['monmap']['mons'])
-            self.gauge(self.NAMESPACE + '.num_mons', num_mons, tags)
-        except KeyError:
-            self.log.debug('Error retrieving mon_status metrics')
+        if self._octopus:
+            try:
+                num_mons = int(raw['status']['monmap']['num_mons'])
+                self.gauge(self.NAMESPACE + '.num_mons', num_mons, tags)
+            except KeyError:
+                self.log.debug('Error retrieving num_mons metric')
+        else:
+            try:
+                num_mons = len(raw['mon_status']['monmap']['mons'])
+                self.gauge(self.NAMESPACE + '.num_mons', num_mons, tags)
+            except KeyError:
+                self.log.debug('Error retrieving mon_status metrics')
 
-        try:
-            num_mons_active = len(raw['mon_status']['quorum'])
-            self.gauge(self.NAMESPACE + '.num_mons.active', num_mons_active, tags)
-        except KeyError:
-            self.log.debug('Error retrieving mon_status quorum metrics')
+            try:
+                num_mons_active = len(raw['mon_status']['quorum'])
+                self.gauge(self.NAMESPACE + '.num_mons.active', num_mons_active, tags)
+            except KeyError:
+                self.log.debug('Error retrieving mon_status quorum metrics')
 
         try:
             stats = raw['df_detail']['stats']
-            self._publish(stats, self.gauge, ['total_objects'], tags)
+            if not self._octopus:
+                self._publish(stats, self.gauge, ['total_objects'], tags)
```

Comment:
Why not publish `total_objects` for ceph octopus? Seems like the metric would be found regardless.

Reply:
This field doesn't exist in the octopus output.